frontend/src/libs/components/input-file/input-file.tsx

```diff
@@ -0,0 +1,125 @@
+import { FIRST_ARRAY_INDEX } from '#libs/constants/constants.js';
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useFormController } from '#libs/hooks/hooks.js';
+import {
+  type FormControl,
+  type FormFieldErrors,
+  type FormFieldName,
+  type FormFieldPath,
+  type FormFieldValues,
+  type FormFieldValuesFromFieldErrors,
+} from '#libs/types/types.js';
+
+import { ErrorMessage, Icon } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
+  errors: FormFieldErrors<T>;
+  fileName: FormFieldPath<T>;
+  fileTypeName: FormFieldPath<T>;
+  fileSizeName: FormFieldPath<T>;
+  label: string;
+  description: string;
+  onChange?: (file: File) => void;
+};
+
+const InputFile = <T extends FormFieldValues>({
+  control,
+  errors,
+  fileName,
+  fileTypeName,
+  fileSizeName,
+  label,
+  description,
+}: Properties<T>): JSX.Element => {
+  const { field: fileField } = useFormController<T>({
+    name: fileName,
+    control,
+  });
+  const { field: fileTypeField } = useFormController<T>({
+    name: fileTypeName,
+    control,
+  });
+  const { field: fileSizeField } = useFormController<T>({
+    name: fileSizeName,
+    control,
+  });
+
+  const fileNaming = (fileField.value as File | null)?.name;
```

Comment:
```suggestion
  const fileName = (fileField.value as File | null)?.name;
```
You can rename the variable from properties to `fileFieldName`

Reply:
Done